managed/CounterStrikeSharp.API/Core/Model/CTakeDamageInfo.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace CounterStrikeSharp.API.Core;
+
+public partial class CTakeDamageInfo
+{
+    /// <summary>
+    /// Retrieves the hitgroup
+    /// </summary>
+    /// <returns>
+    /// Returns a <see cref="HitGroup_t"/> enumeration representing the player's current hit group,
+    /// or <see cref="HitGroup_t.HITGROUP_INVALID"/> if the hit group cannot be determined.
+    /// </returns>
+    public unsafe HitGroup_t HitGroup()
+    {
+        nint v4 = *(nint*)(this.Handle + 0x78);
+
+        if (v4 == nint.Zero)
+        {
+            return HitGroup_t.HITGROUP_INVALID;
+        }
+
+        nint v1 = *(nint*)(v4 + 16);
+
+        if (v1 == nint.Zero)
+        {
+            return HitGroup_t.HITGROUP_GENERIC;
+        }
+
+        return (HitGroup_t)(*(uint*)(v1 + 56));
+    }
```

Comment:
You should be able to convert this to safe code (untested):

```suggestion
    public HitGroup_t HitGroup()
    {
        IntPtr v4 = Marshal.ReadIntPtr(handle, 0x78);

        if (v4 == nint.Zero)
        {
            return HitGroup_t.HITGROUP_INVALID;
        }

        IntPtr v1 = Marshal.ReadIntPtr(v4, 16

        if (v1 == nint.Zero)
        {
            return HitGroup_t.HITGROUP_GENERIC;
        }

        return (HitGroup_t)Marshal.ReadInt32(v1, 56);
    }
```

Reply:
@roflmuffin yes it's working.